textpattern/lib/txplib_misc.php

```diff
@@ -6747,3 +6747,25 @@ function check_file_integrity($flags = INTEGRITY_STATUS)
 
     return $return;
 }
+
+/**
+ * Checks prefs integrity and AutoCreate missing prefs.
+ *
+ */
+
+function check_prefs_integrity()
+{
+    global $prefs;
+    $prefs = get_prefs();
+    include txpath.'/lib/prefs.php';
+
+    foreach ($default_prefs as $event => $event_prefs) {
+        foreach ($event_prefs as $p) {
+            if (! isset($prefs[$p[3]])) {
```

Comment:
Can we not use `set_pref()` or `create_pref()` here instead of a raw insert? We could then take out any (all? most?) `doSlash()` calls in `prefs.php` because we get it done for free in those functions.

Since we're not setting per-user prefs here (so `$txp_user` doesn't matter), and `txplib_db.php` is included in your patch, we might be able to get away with using the prefs API functions. Or have  missed something?